GVRf/Framework/framework/src/main/jni/objects/components/component.h

```diff
@@ -30,31 +30,34 @@ class SceneObject;
 class Component: public HybridObject {
 public:
     Component() :
-            HybridObject(), type_(0), owner_object_() {
+            HybridObject(), type_(0), owner_object_(NULL), enabled_(true) {
     }
 
     Component(long long type) :
-            HybridObject(), type_(type), owner_object_() {
+            HybridObject(), type_(type), owner_object_(NULL), enabled_(true) {
     }
 
     Component(SceneObject* owner_object) :
             type_(0),
+            enabled_(true),
             owner_object_(owner_object) {
     }
 
     Component(long long type, SceneObject* owner_object) :
             type_(type),
+            enabled_(true),
             owner_object_(owner_object) {
     }
 
     virtual ~Component() {
+        set_owner_object(NULL);
     }
 
     SceneObject* owner_object() const {
         return owner_object_;
     }
 
-    void set_owner_object(SceneObject* owner_object) {
+    virtual void set_owner_object(SceneObject* owner_object) {
```

Comment:
Overriding set_owner_object is the way subclasses of Component can do something when they are attached or detached from a scene object. I tound this capability useful when implementing Collider components because I need to keep a special list of them to optimize picking.